### ISSUE TITLE
Monitor SQS queue: Add logging to lambda

### ISSUE THREAD

**Issue by Jimlinz** [linz/geostore#1947]
## User Story

So that we can respond to operational issues with the SQS queue that updates the STAC metadata, as a Geostore admin (devops team member), I want to be alerted of abnormal (and normal) behaviour in the SQS queue.

## Tasks
 - [x] Decide weather to: a) modify the custom Slack app already in the Geostore, or b) use the [existing integration](https://github.com/linz/elasticsearch-config/tree/9ecb2ef89260f75901183786087069548c75d58a/config/li/watcher) between [logs in Elastic](https://li.kb.ap-southeast-2.aws.found.io:9243/s/datalake/app/discover#/?_a=(columns:!(),filters:!(),index:a2b7f020-c9f2-565c-950b-dfcd93e95fd4,interval:auto,query:(language:kuery,query:sqs),sort:!(!('@timestamp',desc)))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-1000y,to:now))) and [Slack](https://li.kb.ap-southeast-2.aws.found.io:9243/s/datalake/app/discover#/?_a=(columns:!(),filters:!(),index:a2b7f020-c9f2-565c-950b-dfcd93e95fd4,interval:auto,query:(language:kuery,query:sqs),sort:!(!('@timestamp',desc)))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-1000y,to:now))). 
- [ ] Add logging to lambda
- [ ] ~~Fix log shipping to elastic~~
- [ ] ~~Set up slack alerts from elastics~~

## Acceptance criteria

Estimate = 8
- [ ] Given there are items on the SQS queue, when the top item in the queue is unable to be processed, then a 'failure' type alert is sent to the #geostore-alerts Slack channel
- [ ] Given there are SQS messsages in the queue, when the SQS queue message is processed, the team is notified via slack of the time it has taken to be processed since it was added to the queue.
- [ ] ~~(can do this later if needed) Given there are items on the SQS queue, when items are being processed slower than expected (start with 15 minutes), then a 'warning' type alert is sent to the #geostore-alerts Slack channel~~
- [ ] Given an item is being processed by the SQS queue, when it is successfully processed, then a 'success' type alert is sent to the #geostore-alerts Slack channel

## Background

The SQS queue for Geostore is configured as as FIFO with no dead letter queue in place. This is likely intentional to ensure that all dataset versions are created and stored correctly. When a message couldn't be processed, it holds up the entire SQS queue. Although an unlikely event, it stops Geostore from updating its catalog. This can be a problem for end-users. 

The Geostore team should be alerted (via Slack?) when a problem like this occurs, and not rely on end users to tell us there is a problem a few days later. This is a bottom of the cliff approach and although a better solution is to ensure all SQS messages that get sent is valid and can be processed, bug #1916 shows that the system isn't fool proof. Bad messages can be created and sent to the queue, even if rarely. 

One way to deal with this is to setup a mechanism that rolls back the entire dataset version creation process when something goes wrong (i.e. SQS message can't be processed, so all the prior tasks performed prior in removed, and end user alerted). This could be challenging to put in place, and likely a huge task. Setting up an alerting mechanism could be an interim solution. 

## How to Reproduce abnormal behavious in the queue

Update dataset version with `item.json` rather than `collection.json` or `catalog.json`. This triggers the step function and complete without a problem; however, it sends a bad SQS message that couldn't be processed and subsequently holds up the entire queue. 

- Ref #1916 

#### Definition of Done

-  [ ] This bug is **done**:
   -  [ ] Bug resolved to **user's** satisfaction
   -  [ ] Automated tests are passing
   -  [ ] Code is peer reviewed and pushed to master
   -  [ ] Deployed successfully to test environment
   -  [ ] Checked against
          [CODING guidelines](https://github.com/linz/geostore/blob/master/CODING.md)
   -  [ ] Relevant new tasks are added to backlog and communicated to the team
   -  [ ] Important decisions recorded in the issue ticket
   -  [ ] Readme/Changelog/Diagrams are updated
   -  [ ] Product Owner has approved as complete
   -  [ ] No regression to functional or
          [non-functional](https://github.com/linz/geostore/blob/master/.github/ISSUE_TEMPLATE/user_story.md)
          requirements

**Comment by billgeo** [comment]
Hey @Jimlinz . Thanks for this. A reasonable priority I think. I reworded this as a user story for monitoring. We have the other issue to fix that one bug, and we should discuss what else we want to do now, like maybe adding more tests to catch potential bugs, or not. I take your point that there is possibly more we want to do here though, So left all that background info in here.

Let me know if I have misrepresented what you were trying to say here and we can adjust it.
